requests_kerberos/kerberos_.py

```diff
@@ -285,6 +284,21 @@ def deregister(self, response):
         response.request.deregister_hook('response', self.handle_response)
 
     def __call__(self, request):
+        if self.force_preemptive:
+            # add Authorization header before we receive a 401
+            # by the 401 handler
+            host = urlparse(request.url).hostname
+
+            auth_header = self.generate_request_header(None, host, is_preemptive=True)
+
+            if auth_header is None:
+                # GSS setup failure, return error and details, if any
+                raise KerberosExchangeError("Kerberos GSS Exchange failed: %s" % str(sl.stored_logs))
```

Comment:
There are no early returns in `generate_request_header` anymore, so I don't think we can hit this.